docs/guide/interactions/views.rst

```diff
@@ -0,0 +1,467 @@
+However, the arguments passed to the function are different. Rather than ``self`` referring to the component, it refers to the view, with
+the component being passed as the last argument.
+
+When using component decorators, we no longer need to explicitly add the component to the view, this is done automatically.
+
+
+Sending views
+--------------
+
+We send views using the ``view`` parameter in methods which send messages, for example :meth:`TextChannel.send`.
+
+In our case we're creating a confirmation prompt for our ban command, so we'll want to use the :meth:`InteractionResponse.send_message` method.
+
+Our command might look like this:
+
+.. code-block:: python
+
+    @discord.app_commands.command()
+    @discord.app_commands.describe(member='The member to ban.')
+    async def ban(interaction: discord.Interaction, member: discord.Member) -> None:
+        """Ban a member from the server."""
+        confirmation = Confirm(interaction.user)
+        await interaction.response.send_message(f'Are you sure you want to ban {member.name}?', view=confirmation)
+        await confirmation.wait()
+        if confirmation.result:
+            await member.ban()
+
+We first assign the ``View`` instance to the variable ``confirmation``, because we'll need it later.
+In our interaction response we send the instance via the ``view`` parameter.
+Finally, using :meth:`confirmation.wait() <discord.ui.View.wait>`, we wait for the view to stop listening for interactions,
+which occurs either when the user clicks on a button (as we had called :meth:`View.stop() <discord.ui.View.stop>`)
+or the view had timed-out.
+
+Since our component callbacks assign the ``result`` attribute of the view, we can use it to determine if the user clicked on the
+`Yes` or `No` button, and ban them if they clicked `Yes`.
+
+
+Persistent Views
+-----------------
+
+There are instances where we might want to create a view that will persist for a long time.
+For example, a view which allows members of a Guild to select a role to assign to themselves.
+
+.. image:: /images/guide/interactions/view_select_example.png
+
+Let's make one such view.
+
+
+Designing a Persistent View
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To start off, we need to create a class to represent the view, which we'll achieve by creating a 
+subclass of :class:`~discord.ui.View`. In that class we need a :class:`~discord.ui.Select` component,
+which is a dropdown menu using the :func:`@select <discord.ui.select>` decorator. We can set some 
+placeholder text for the dropdown menu and set both the minimum and maximum number of elements 
+a user can select to `1`.
+
+To start out, our code might look like this:
+
+.. code-block:: python
+
+    class RoleSelector(discord.ui.View):
+        def __init__(self) -> None:
+            super().__init__(timeout=None)
+
+        @discord.ui.select(
```

Comment:
`discord.ui.select` has an kwarg called `cls`. In this, you pass in the type of select that you want. In this case, all we need to do is to pass in `discord.ui.RoleSelect`. The current implementation in this page relies on an old method on doing this and uses the base select type.

Reply:
`RoleSelect` doesn't provide the same functionality though (it lists all roles). If we'd rather use it then it would make more sense to just replace the entire example.

Reply:
It would make more sense instead of roles, that they are something unique such as colors imo. this kinda depends if the example should be replaced or not

Reply:
I fully agree with Josh on this one. The <cls>Select classes provide no filter (which is a bit of a silly move on Discord's side IMO>. As such I still view this as best practice as we can filter what is shown to the user, and not make an accidental footgun where a user could give themselves an admin role.